tools/lsp/main.rs

```diff
@@ -127,10 +130,21 @@ fn main() {
 
 pub fn run_lsp_server() -> Result<(), Error> {
     let (connection, io_threads) = Connection::stdio();
-    let capabilities = server_loop::server_capabilities();
-    let server_capabilities = serde_json::to_value(&capabilities).unwrap();
-    let initialization_params = connection.initialize(server_capabilities)?;
-    main_loop(&connection, initialization_params)?;
+    let (id, params) = connection.initialize_start()?;
+
+    let initialization_reply = InitializeResult {
+        capabilities: server_loop::server_capabilities(),
+
+        server_info: Some(ServerInfo {
+            name: "slint-ls".into(),
```

Comment:
I suggest to use the name from cargo as well:

```suggestion
            name: env!("CARGO_PKG_NAME").to_string() ,
```

We don't use "slint-ls" anywhere.  Where does it comes from.  Are you sure the pasted json comes from the current server and not something that is filled by your test system, or so?

Reply:
Good idea!